python/tchannel/exceptions.py

```diff
@@ -34,18 +34,21 @@ class InvalidChecksumException(TChannelException):
     pass
 
 
+class StreamingException(TChannelException):
+    """Represent Streaming Message Exception"""
+    pass
+
+
 class TChannelApplicationException(TChannelException):
     """The remote application returned an exception.
 
     This is not a protocol error. This means a response was received with the
     ``code`` flag set to fail.
     """
-    def __init__(self, code, arg_1, arg_2, arg_3):
+    def __init__(self, code, args):
         super(TChannelException, self).__init__(
-            'TChannel application error (%s, %s, %s)' % (arg_1, arg_2, arg_3)
+            'TChannel application error (%s)' % (args)
```

Comment:
Is args a tuple? If so, this will fail because `(args)` just means `args`, not a tuple with `args`.

    In [1]: "foo %s" % ('a', 'b', 'c')
    ---------------------------------------------------------------------------
    TypeError                                 Traceback (most recent call last)
    <ipython-input-1-222a695f910b> in <module>()
    ----> 1 "foo %s" % ('a', 'b', 'c')

    TypeError: not all arguments converted during string formatting

You probably want to use the `(x,)` form to create a tuple with a single item:

    In [2]: "foo %s" % (('a', 'b', 'c'),)
    Out[2]: "foo ('a', 'b', 'c')"